snykTask/src/task-args.ts

```diff
@@ -102,31 +111,45 @@ class TaskArgs {
     return this.projectName;
   }
 
+  // validate based on testTypeSeverityThreshold applicable thresholds
   public validate() {
-    if (this.failOnThreshold) {
-      if (this.isNotValidThreshold(this.failOnThreshold)) {
-        const errorMsg = `If set, failOnThreshold must be '${Severity.CRITICAL}' or '${Severity.HIGH}' or '${Severity.MEDIUM}' or '${Severity.LOW}' (case insensitive). If not set, the default is '${Severity.LOW}'.`;
-        throw new Error(errorMsg);
-      }
+    const taskTestType = this.testType || TestType.APPLICATION;
+    const validTestTypes: any = Object.values(TestType);
+    const taskTestTypeThreshold =
+      testTypeSeverityThreshold.get(taskTestType) ??
+      testTypeSeverityThreshold.get(TestType.APPLICATION);
```

Comment:
As far as I understand, in this block:

```ts
    const taskTestType = this.testType || TestType.APPLICATION;
    const taskTestTypeThreshold =
      testTypeSeverityThreshold.get(taskTestType) ??
      testTypeSeverityThreshold.get(TestType.APPLICATION);
```

when we check `testTypeSeverityThreshold.get(taskTestType)`, `taskTestType` is already `this.testType` or `TestType.APPLICATION`.

Could it be that only writing `const taskTestTypeThreshold = testTypeSeverityThreshold.get(taskTestType)` is enough?

But maybe the problem comes from the default value of `testType` defined at the beginning of the class?
```ts
class TaskArgs {
  testType: string | undefined = '';
```
I can see that if `this.testType` is an empty string, which isn't a key in `testTypeSeverityThreshold`, then the nullish coalescing operator is actually necessary...

Reply:
If I understand the README correctly, it seems that `testType` is by default `"app"`, right? Maybe we could initialise it to `"app"` instead of empty string?

Reply:
yes, the declaration of `testType` could be explicitly defaulted to `app`.  The null coalescing operator is required because current behaviour is this azure extension Task still runs properly with all other provided valid args applicable for a `app` test even when `testType` is itself set invalid.